jetty-server/src/main/java/org/eclipse/jetty/server/handler/InetAccessHandler.java

```diff
@@ -123,23 +181,30 @@ public void handle(String target, Request baseRequest, HttpServletRequest reques
     /**
      * Checks if specified address and request are allowed by current InetAddress rules.
      *
-     * @param address the inetAddress to check
-     * @param request the request to check
+     * @param addr     the inetAddress to check
+     * @param baseRequest the base request to check
+     * @param request     the HttpServletRequest request to check
      * @return true if inetAddress and request are allowed
      */
-    protected boolean isAllowed(InetAddress address, HttpServletRequest request)
+    protected boolean isAllowed(InetAddress addr, Request baseRequest, HttpServletRequest request)
     {
-        boolean allowed = _set.test(address);
+        String name = baseRequest.getHttpChannel().getConnector().getName();
         if (LOG.isDebugEnabled())
-            LOG.debug("{} {} {} for {}", this, allowed ? "allowed" : "denied", address, request);
-        return allowed;
+        {
+            Boolean allowedByName = _names.isIncludedAndNotExcluded(name);
+            Boolean allowedByAddr = _addrs.isIncludedAndNotExcluded(addr);
+            LOG.debug("{} allowedByName={} allowedByAddr={} for {}/{}", this, allowedByName, allowedByAddr, addr, request);
+        }
+        return IncludeExclude.and(_addrs, addr, _names, baseRequest.getHttpChannel().getConnector()::getName);
```

Comment:
Too difficult to understand. See below.
This should be:
```java
return _addrs.isAllowed(addr) && _names.isAllowed(connectorName);
```
Just add a strict `boolean isAllowed()` to `IncludeExcludeSet` and we're done.

Reply:
But that is not the correct logic for combine to IncludeExclude sets.   There is a difference between being include, excluded or neither when combining results.